component/componenttest/nop_exporter.go

```diff
@@ -26,11 +26,9 @@ import (
 // nopExporterFactory is factory for nopExporter.
 type nopExporterFactory struct{}
 
-var nopExporterFactoryInstance = &nopExporterFactory{}
-
 // NewNopExporterFactory returns a component.ExporterFactory that constructs nop exporters.
 func NewNopExporterFactory() component.ExporterFactory {
-	return nopExporterFactoryInstance
+	return &nopExporterFactory{}
```

Comment:
Why all the changes related to these being singletons?

Reply:
I had referenced a similar change in the `consumererror` package, but I realize that the usage there is different. So I will revert to the previous implementation.